lib/address.py

```diff
@@ -255,11 +256,15 @@ def to_ui_string(self, hex_only = False):
                     def lookup(x):
                         return OpCodes.reverseLookup.get(x, ('('+str(x)+')'))
                     if isinstance(op, tuple):
-                        ret += lookup(op[0]) + " " + op[1].decode('utf-8')
+                        try:
+                            opData = op[1].decode('utf-8')
+                        except:
+                            opData = op[1].hex() if isinstance(op[1], bytes) else str(op[1])
```

Comment:
Aren't they always bytes (and below)?  We should have a consistent data model.

Reply:
That code is a nightmare to me. It sometimes returns tuples, sometimes ints, sometimes bytes.

Reply:
It's our code, all in our control.  It should only be doing one thing.  I find it frustrating when the code is made more spaghetti-like (and Electrum is just that) with superstitious code rather than just ensuring it's well-defined and enforcing it.

Reply:
Your frustrations are noted and in this case empathized with.  However I am not going to go in and rewrite that function since it's called in exactly 1 place and it doesn't matter that much.

I need to focus on other stuff I'm working on.

I agree with you for the record.